pilot/cmd/pilot-agent/main.go

```diff
@@ -290,6 +294,24 @@ var (
 				defer stsServer.Stop()
 			}
 
+			// Start a local DNS server on 15053, forwarding to DNS-over-TLS server
+			// This will not have any impact on app unless interception is enabled.
+			// We can't start on 53 - istio-agent runs as user istio-proxy.
+			// This is available to apps even if interception is not enabled.
+
+			// TODO: replace hardcoded .global. Right now the ingress templates are
+			// hardcoding it as well, so there is little benefit to do it only here.
+			if dns.DNSTLSEnableAgent.Get() {
+				// In the injection template the only place where global.proxy.clusterDomain
+				// is made available is in the --domain param.
+				// Instead of introducing a new config, use that.
+
+				dnsSrv := dns.InitDNSAgent(proxyConfig.DiscoveryAddress,
+					role.DNSDomain, sa.RootCert,
+					[]string{".global."})
+				dnsSrv.StartDNS(dns.DNSAgentAddr, nil)
+			}
+
```

Comment:
So what’s the port we are going to listen on ? And the iptables redirection for that port?

Reply:
In agent - 15013 - I checked and it's not in use. I'm trying to keep the agent ports in the same range (15010, etc). Most of the ports are currently hardcoded in various parts of istio - I don't think it's worth the effort to make them configurable. For example the capture port for the inbound listener.